Core/Core/Dashboard/K5/ViewModel/Grades/K5GradesViewModel.swift

```diff
@@ -16,13 +16,112 @@
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 //
 
-class K5GradesViewModel {
+public class K5GradesViewModel: ObservableObject {
 
+    @Published public private(set) var grades: [K5GradeCellViewModel] = []
+    @Published public private(set) var gradingPeriods: [K5GradingPeriod] = []
+    @Published public private(set) var currentGradingPeriod: K5GradingPeriod
+
+    private let env = AppEnvironment.shared
+    private var studentID = ""
+    private lazy var courses = env.subscribe(GetUserCourses(userID: studentID)) { [weak self] in
+        self?.coursesUpdated()
+    }
+    private let defaultCurrentGradingPeriod = K5GradingPeriod(periodID: nil, title: "Current Grading Period")
+
+    // MARK: Refresh
+    private var refreshCompletion: (() -> Void)?
+    private var forceRefresh = false
+
+    init() {
+        studentID = env.currentSession?.userID ?? ""
+        currentGradingPeriod = defaultCurrentGradingPeriod
+        courses.refresh()
+    }
+
+    private func coursesUpdated() {
+        grades.removeAll()
+        gradingPeriods = [defaultCurrentGradingPeriod]
+        grades = courses.filter({ !$0.isHomeroomCourse }).map {
```

Comment:
Since `grades` are assigned to a new value, do we need `grades.removeAll()` two lines before?